spring-cloud-gcp-native-support/src/main/java/com/google/cloud/spring/support/nativex/SpringDataFirestoreDocumentsComponentProcessor.java

```diff
@@ -0,0 +1,78 @@
+package com.google.cloud.spring.support.nativex;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.nativex.domain.reflect.FieldDescriptor;
+import org.springframework.nativex.hint.AccessBits;
+import org.springframework.nativex.type.*;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Native Component processor adding reflection support for classes annotated with
+ * {@link com.google.cloud.spring.data.firestore.Document}. These classes will be found by spring aot if
+ * <code>spring-context-indexer</code> was used to index such classes.
+ *
+ * @see <a href="https://github.com/spring-projects-experimental/spring-native/blob/832991d57686627b06792d55555cb9497475b3c5/spring-native-configuration/src/main/java/org/springframework/data/JpaComponentProcessor.java#L34">JpaComponentProcessor</a>
+ */
+public class SpringDataFirestoreDocumentsComponentProcessor implements ComponentProcessor {
+    private static final Log logger = LogFactory.getLog(SpringDataFirestoreDocumentsComponentProcessor.class);
+
+    private final TypeProcessor typeProcessor = new TypeProcessor(
```

Comment:
Can `typeProcessor` implementation and the associated logic be replaced with the built-in `TypeProcessor.namedProcessor()`? The functionality is largely similar.

Reply:
applied! thanks 🙏